tests/tck/features/yield/parameter.feature

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2021 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+Feature: Parameter
+
+  Background:
+    Given a graph with space named "nba"
+    Given parameters: {"p1":1,"p2":true,"p3":"Tim Duncan","p4":3.3}
+
+  Scenario: return parameters
+    When executing query:
+      """
+      RETURN abs($p1)+1 AS ival, $p2 and false AS bval, $p3+"ef" AS sval, round($p4)+1.1 AS fval
+      """
+    Then the result should be, in any order:
+      | ival | bval  | sval           | fval |
+      | 2    | false | "Tim Duncanef" | 4.1  |
+
+  Scenario: match with parameters
+    When executing query:
+      """
+      MATCH (v:player)-[:like]->(n) WHERE id(v)==$p3 and n.age>$p1+29
+      RETURN n.name AS dst
+      """
+    Then the result should be, in any order:
+      | dst             |
+      | "Manu Ginobili" |
+      | "Tony Parker"   |
+
+  Scenario: var with parameters
+    When executing query:
+      """
+      $p1=GO FROM "Tim Duncan" OVER like WHERE like.likeness>$p1 yield like._dst as dst;
+      GO FROM $p1.dst OVER like YIELD DISTINCT $$.player.name AS name
```

Comment:
```
 $p1=GO FROM "Tim Duncan" OVER like WHERE like.likeness>$p1 yield like._dst as dst;
 GO FROM $p1.dst OVER like Where like.age > $p1 YIELD DISTINCT $$.player.name AS name.
```
What is the behavior of this test case?
We need more test cases.

Reply:
`$p1` started as a parameter and was rewritten as a variable, just consistent with what we talked about earlier.
---

Such as what test cases？

Reply:
Where did you do the rewriting?

Reply:
Expression rewriting?

Reply:
The implementation layer is the same as `VariableExpression`, it is based on `valueMap_`, so there is no need to do rewriting. `valueMap_` is automatically written to at runtime.

Reply:
> ```
>  $p1=GO FROM "Tim Duncan" OVER like WHERE like.likeness>$p1 yield like._dst as dst;
>  GO FROM $p1.dst OVER like Where like.age > $p1 YIELD DISTINCT $$.player.name AS name.
> ```
> 
> What is the behavior of this test case? We need more test cases.

Variables and parameters with the same name are not supported for the time being, just consistent with our offline discussion.